multiversx_sdk_core/transaction_factories/smart_contract_factory.py

```diff
@@ -0,0 +1,192 @@
+from pathlib import Path
+from typing import Any, List, Optional, Protocol, Tuple
+
+from Cryptodome.Hash import keccak
+
+from multiversx_sdk_core.address import Address
+from multiversx_sdk_core.constants import (ARGS_SEPARATOR,
+                                           CONTRACT_DEPLOY_ADDRESS,
+                                           DEFAULT_HRP,
+                                           TRANSACTION_OPTIONS_DEFAULT,
+                                           TRANSACTION_VERSION_DEFAULT,
+                                           VM_TYPE_WASM_VM)
+from multiversx_sdk_core.interfaces import (IAddress, IChainID, IGasLimit,
+                                            IGasPrice, INonce,
+                                            ITransactionValue)
+from multiversx_sdk_core.serializer import arg_to_string
+from multiversx_sdk_core.transaction import Transaction
+from multiversx_sdk_core.transaction_payload import TransactionPayload
+from multiversx_sdk_core.utils import read_binary_file
+
+
+class IConfig(Protocol):
+    chain_id: IChainID
+
+
+class CodeMetadata:
```

Comment:
Now I see, we have an existing implementation here:

https://github.com/multiversx/mx-sdk-py-core/blob/main/multiversx_sdk_core/code_metadata.py

Reply:
Indeed. I didn't see it earlier. I will use it.